src/main/java/by/onliner/page/components/ShoppingCartPopup.java

```diff
@@ -0,0 +1,43 @@
+package by.onliner.page.components;
+
+import by.onliner.page.AbstractComponent;
+import by.onliner.util.Waiter;
+import org.awaitility.Awaitility;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class ShoppingCartPopup extends AbstractComponent {
+
+    @FindBy(className = "product-recommended__sidebar-close")
+    private WebElement shoppingCartPopupCloseButton;
+
+    public ShoppingCartPopup() {
+        super();
+    }
+
+    @Override
+    public void waitForComponentOpened() {
+        Waiter.elementToBeClickable(shoppingCartPopupCloseButton);
+    }
+
+    public boolean visibilityOfShoppingCartPopup() {
+        try {
+            Awaitility.await()
+                    .atMost(5, SECONDS)
+                    .atLeast(1, SECONDS)
+                    .pollInterval(1, SECONDS)
+                    .until(() -> shoppingCartPopupCloseButton.isDisplayed());
+            return true;
+        } catch (Exception e) {
```

Comment:
If you are going to use an exception here please narrow it down. Also check whether you need that exception in this method, in what cases an exception can be thrown when checking whether an element is displayed?

Reply:
*

Reply:
⭐

Reply:
Fixed